lib/superstore/types/range_type.rb

```diff
@@ -0,0 +1,45 @@
+module Superstore
+  module Types
+    class RangeType < BaseType
+      class_attribute :subtype
+
+      def encode(range_hash)
+        [
+          convert(:encode, range_hash[:lower]),
+          convert(:encode, range_hash[:upper])
+        ]
+      end
+
+      def decode(range_tuple)
+        {
+          lower: convert(:decode, range_tuple[0]),
+          upper: convert(:decode, range_tuple[1])
+        }
+      end
+
+
+      def typecast(value)
+        {
+          lower: convert(:typecast, value[:lower] || value['lower']),
+          upper: convert(:typecast, value[:upper] || value['upper'])
+        }
+      end
+
+      private
+
+      def convert(method, value)
+        subtype.send(method, value) unless value.nil?
```

Comment:
`unless value.nil?` is unnecessary IMO.

Reply:
That requires all types to check for nil in all methods, which is currently not required.